src/cmd-buildextend-metal

```diff
@@ -217,10 +209,17 @@ cat >image-dynamic.json << EOF
 }
 EOF
 cat "${image_json}" image-dynamic.json | jq -s add > image-for-disk.json
+platforms_json=
+if [ -e "${configdir}/platforms.yaml" ]; then
+    platforms_json="${workdir}/tmp/platforms.json"
+    yaml2json "${configdir}/platforms.yaml" "${platforms_json}"
+fi
 runvm "${target_drive[@]}" -- \
         /usr/lib/coreos-assembler/create_disk.sh \
             --config "$(pwd)"/image-for-disk.json \
             --kargs "\"${kargs}\"" \
+            --platform "${ignition_platform_id}" \
+            ${platforms_json:+--platforms-json "${platforms_json}"} \
```

Comment:
I think it'd be cleaner to just add to `disk_args` in the if-statement above.

Reply:
Fair.  Once we're done ratcheting `platforms.yaml` into the config, this argument will be deconditionalized in a followup PR, so I'm inclined to leave as is to reduce churn.

Reply:
Legacy removal in https://github.com/coreos/coreos-assembler/pull/2899.